BSP/STM32F413/Src/BSP_Contactor.c

```diff
@@ -2,41 +2,110 @@
 
 #include "BSP_Contactor.h"
 #include "stm32f4xx.h"
+#include "stm32f4xx_gpio.h"
+#include "stm32f4xx_rcc.h"
 
 /**
- * @brief   Initializes the GPIO pins that interfaces with the Contactor.
+ * Used pins:
+ * A:
+ * 		PA2, PA3 - ADC and UART
+ * 		PA8 - I2C SCL
+ * 		PA4 thru PA7 - Lights
+ * 		PA15 - SPI CS
+ * B:
+ * 		PB3 thru PB5 - SPI
+ * 		PB8, PB9 - CAN
+ * 		PB10 - UART
+ * 		PB12 - Lights
+ * 		PB14, PB15 - Fans
+ * C:
+ * 		PC5 - also UART
+ * 		PC6, PC7 - Fans
+ * 		PC9 - I2C SDA
+ * 		PC0 thru PC4 - Lights
+ * 		PC10 thru PC12 - more SPI
+ * D:	
+ * 		PD2 - SPI CS
+ * 		
+ * 
+ * 		Available Pins for contactors:
+ * 		PB0-1 used for C1
+ * 		PC0-1 can use for C3
+ * 		PA4-5 can use for C2
+ */
+
+/**
+ * @brief   Initializes the GPIO pins that interfaces with the Contactors.
  *          Two GPIO pins are initialized. One as an output and one as an input.
  *          The output pin controls the state and the input pin views what state the contactor is through the
  *          Aux pins.
  * @param   None
  * @return  None
  */
 void BSP_Contactor_Init(void) {
-    GPIO_InitTypeDef GPIO_InitStruct;
+    GPIO_InitTypeDef GPIO_C1Init;
+	GPIO_InitTypeDef GPIO_C2Init;
+	GPIO_InitTypeDef GPIO_C3Init;
 	
 	// Initialize clock
-	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE);
+	RCC_AHB1PeriphClockCmd(C1_PERIPH, ENABLE);
+	RCC_AHB1PeriphClockCmd(C2_PERIPH, ENABLE);
+	RCC_AHB1PeriphClockCmd(C3_PERIPH, ENABLE);
 	
-	// Initialize PB0 as output
-	GPIO_InitStruct.GPIO_Pin = GPIO_Pin_0;			// Pin 0 is output
-	GPIO_InitStruct.GPIO_Mode = GPIO_Mode_OUT;
-	GPIO_InitStruct.GPIO_Speed = GPIO_Speed_50MHz;
-	GPIO_InitStruct.GPIO_OType = GPIO_OType_PP;
-	GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_UP;
-	GPIO_Init(GPIOB, &GPIO_InitStruct);
-	GPIO_InitStruct.GPIO_Pin = GPIO_Pin_1;			// Pin 1 is input
-	GPIO_InitStruct.GPIO_Mode = GPIO_Mode_IN;
-	GPIO_Init(GPIOB, &GPIO_InitStruct);
+	// Initialize Port B for Contactor 1
+	// PB0 - output
+	// PB1 - input
+	GPIO_C1Init.GPIO_Pin = GPIO_Pin_0;
```

Comment:
could make the gpio init block a function and call it three times with different i/o pins